lmfit/models.py

```diff
@@ -983,6 +984,57 @@ def guess(self, data, x=None, negative=False, **kwargs):
     guess.__doc__ = COMMON_GUESS_DOC
 
 
+class ThermalDistributionModel(Model):
+    r"""Return a thermal distribution function. Variable ``kind`` defines
+    the kind of distribution as below with parameters ``amplitude``
+    (:math:`A`), ``center`` (:math:`x_0`) and ``kT`` (:math:`kT`).
+
+    .. math::
+
+        f(x; A, x_0, kT, T) = \frac{1}{A \exp(\frac{x - x_0}{kT}) + kind}
+
+    As such, these thermal distributions are available:
+
+    - Bose-Einstein distribution via :math:`kind=-1` (the default)
+    - Maxwell-Boltzmann distribution via :math:`kind=0`
+    - Fermi-Dirac distribution via :math:`kind=1`
+
+    see http://hyperphysics.phy-astr.gsu.edu/hbase/quantum/disfcn.html
+
+    Comments:
+
+    - Amplitude and kind are set not to vary by default.
+    - kT should be defined in the same units as x (kB = 8.617e-5 eV/K).
+    - set :math:`kT<0` to implement the energy loss convention common in
+      scattering research.
+    """
+    def __init__(self, independent_vars=['x'], prefix='', nan_policy='raise',
+                 **kwargs):
+        kwargs.update({'prefix': prefix, 'nan_policy': nan_policy,
+                       'independent_vars': independent_vars})
+        super().__init__(thermal_distribution, **kwargs)
+        self._set_paramhints_prefix()
+
+    def _set_paramhints_prefix(self):
+        self.set_param_hint('amplitude', min=0, max=1, vary=False)
+        self.set_param_hint('kind', min=-1, max=1, vary=False)
+
+    def guess(self, data, x=None, negative=False, **kwargs):
+        """Estimate initial model parameter values from data."""
+        if x is None:
+            center = 0
+            kT = 1
+        else:
+            center = np.mean(x)
+            kT = (max(x) - min(x))/10
+
```

Comment:
sorry, but I am not sure I understand the guessing here. Why would be ```np.mean(x)``` a good estimate of ```center```? Let's say I do the following:
```
x = np.linspace(0, 100, 1000)
y = lmfit.lineshapes.thermal_distribution(x, amplitude=1.0, center=10, kT=1.0, kind=-1)
``` 
in this case the ```center``` will be at 10, but your guess function sets it to 50. Wouldn't it be better to re-use some of the code from ```guess_from_peak``` instead? For example, 
```
imaxy = index_of(data, max(data))
center = x[imaxy]
```
that will result in ```center=10.010```.

Along the same lines, I'm also not sure why ```(max(x) - min(x))/10``` would be a good estimate for ```kT```; can you elaborate on that?

Reply:
Right now the guesses of the center and kt parameters are based on something similar to what was done for the step function. This is a very crude guess of just setting these to the mean of x and 1/10 of the range of x. These are the only guesses I can think of that are at all sensible for all three forms. Some more sophisticated guesses could be constructed for the bose and Fermi forms, for which the center should correspond to the maximum in the absolute value of the derivative of x. This would depend on how cleaver we try to be?